apps/octoprint/octoprint.star

```diff
@@ -0,0 +1,303 @@
+                                transforms = [animation.Translate(-0, 0)],
+                                curve = "ease_in_out",
+                            ),
+                            animation.Keyframe(
+                                percentage = 1.0,
+                                transforms = [animation.Translate(64, 0)],
+                                curve = "ease_in_out",
+                            ),
+                        ],
+                    ),
+                    animation.Transformation(
+                        child = render.Box(
+                            width = 64,
+                            height = 32,
+                            color = "#000000",
+                            child = render.Image(src = snapshot, width = 64, height = 32),
+                        ),
+                        duration = 200,
+                        delay = 180,
+                        origin = animation.Origin(0, 0),
+                        keyframes = [
+                            animation.Keyframe(
+                                percentage = 0.0,
+                                transforms = [animation.Translate(-64, 0)],
+                                curve = "ease_in_out",
+                            ),
+                            animation.Keyframe(
+                                percentage = 0.1,
+                                transforms = [animation.Translate(-0, 0)],
+                                curve = "ease_in_out",
+                            ),
+                            animation.Keyframe(
+                                percentage = 1.0,
+                                transforms = [animation.Translate(0, 0)],
+                                curve = "ease_in_out",
+                            ),
+                        ],
+                    ),
+                ],
+            ),
+        )
+
+    else:
+        return render.Root(
+            child = render.Box(
+                width = 64,
+                height = 32,
+                child = render.Padding(
+                    pad = (0, 1, 0, 0),
+                    child = render.Column(
+                        main_align = "center",
+                        cross_align = "center",
+                        expanded = True,
+                        children = [
+                            render.Text("%s" % name, font = "tom-thumb"),
+                            render.Text(state, color = stateColor),
+                            render.Text("%s hrs left" % printTimeLeft),
+                            renderProgress("Completion", completion, 1),
+                        ],
+                    ),
+                ),
+            ),
+        )
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "serverIP",
+                name = "Server IP",
+                desc = "Ex: (192.168.1.123)",
```

Comment:
Since apps run on Tidbyt servers, the app won't be able to reach local networks. I suspect that may render this app unusable in most cases?

Reply:
This (like the roomba app), again requires you to port forward your octoprint port and use the public IP. I'll change the desc here so it's not misleading

Reply:
Got it. I should also mention that these field descriptions strings unfortunately aren't exposed to users in the mobile app, so it might be good to mention it in the app description as well.

Reply:
Fixed